handler/push.go

```diff
@@ -207,20 +200,20 @@ func LegacyPush(
 	}
 }
 
-// sanitizeLabels ensures that all the labels in groupingLabels and
-// requiredLabels are present in each MetricFamily in metricFamilies. The label
-// values from groupingLabels are set in each MetricFamily, no matter what. In
-// contrast, requiredLabels contains only label names and no label values. Only
-// if any of those labels are not present at all in a MetricFamily, they will be
-// created (with an empty string as value).
+// sanitizeLabels ensures that all the labels in groupingLabels and the
+// `instance` label is present in each MetricFamily in metricFamilies. The
+// label values from groupingLabels are set in each MetricFamily, no matter
+// what. In contrast, 'instance' will have no label values. Only if any of
+// those labels are not present at all in a MetricFamily, they will be created
+// (with an empty string as value).
 //
 // Finally, sanitizeLabels sorts the label pairs of all metrics.
 func sanitizeLabels(
 	metricFamilies map[string]*dto.MetricFamily,
-	groupingLabels map[string]string, requiredLabels map[string]struct{},
+	groupingLabels map[string]string,
 ) {
 	gLabelsNotYetDone := make(map[string]string, len(groupingLabels))
-	rLabelsNotYetDone := make(map[string]struct{}, len(requiredLabels))
+	rLabelsNotYetDone := make(map[string]struct{}, 1)
```

Comment:
We don't really need a map here anymore. It's only the check for instance now, so the logic below should be simplified accordingly.